scripts/xtensa-build-zephyr.py

```diff
@@ -962,7 +963,8 @@ def install_lib(sof_lib_dir, abs_build_dir, platform_wconfig):
 			llext_input = entry_path / (llext_base + '.llext')
 			llext_output = entry_path / (llext_file + '.ri')
 
-			sign_cmd = [platform_wconfig.get("rimage.path"), "-o", str(llext_output),
+			rimage_cmd = re.sub(r"[\"']", "", platform_wconfig.get("rimage.path"))
```

Comment:
This needs at the very least an explanation.

`zephyr/scripts/west_commands/sign.py` has never needed this sort of regex hack and it has been signing SOF in SOF CI for ages on both Linux and Windows.

So why should `xtensa-build-zephyr.py` need it?

Reply:
@marc-hb without this I was getting errors of not being able to find the file with some wild-looking multilevel quoting containing both double and single quotes surrounding the name. So apparently the parser decided that some of those quotes were a part of the name and failed to find that file. Would such a comment be enough or do you need a specific example of what that name and error looked like? I'd probable need to re-run a test to recreate it.

Reply:
> without this I was getting errors of not being able to find the file with some wild-looking multilevel quoting containing both double and single quotes surrounding the name. 

Please file a bug. Then you can just reference the bug here.

Reply:
> zephyr/scripts/west_commands/sign.py has never needed this sort of regex hack and it has been signing SOF in SOF CI for ages on both Linux and Windows.

Oh wait: there is an obvious difference: `execute_command()`. Maybe there is a bug in it. Can you try temporarily bypassing it and calling whichever `subprocess` call is inside `execute_command()` directly? Just for testing purposes.

Reply:
This hack is not even mentioned in the message of the commit 008a9a19175bc64ad8b it belongs to... so is this a Windows-only issue?

Reply:
> This hack is not even mentioned in the message of the commit [008a9a1](https://github.com/thesofproject/sof/commit/008a9a19175bc64ad8b7969202610d438bbad793) it belongs to... so is this a Windows-only issue?

yes

Reply:
> Oh wait: there is an obvious difference: `execute_command()`. Maybe there is a bug in it. Can you try temporarily bypassing it and calling whichever `subprocess` call is inside `execute_command()` directly? Just for testing purposes.

@marc-hb didn't help https://github.com/thesofproject/sof/actions/runs/9792315080/job/27037914711?pr=9281

Reply:
ok, obviously that couldn't help and the problem isn't in `execute_command()` since I'm cleaning up the string of superfluous (see, how superfluous something can be fatal) quotes **before** calling it, all those quote are there already when extracting the path from `platform_wconfig`.

Reply:
I'm fine with this now we have an explanation and inline comment. This will allow the developers who use Windows to come up with a better solution when time permits.

Reply:
I found a Windows system and got to the bottom of this. Please use the following code, it should work both on Linux and Windows (obviously: try it on your workstation first)

```python
            # See why the shlex() parsing step is required at
            # https://docs.zephyrproject.org/latest/develop/west/sign.html#rimage
            # and in Zephyr commit 030b740bd1ec
			rimage_cmd = shlex.split(platform_wconfig.get('rimage.path'))[0]
```

Despite having designed and written all this, I completely forgot about it...

> Oh wait: there is an obvious difference: execute_command()

I found the actual difference: `config_get()` is not the same as `config.get()`. That was the clue.